index.js

```diff
@@ -108,7 +108,30 @@ function resizeImage (filePath, requestData, data, options, cb) {
                 var width, height;
                 var size = metadata.size;
 
-                if (options.scaleUp || (!options.scaleUp && ( size.width > requestData.width || size.height > requestData.height))) {
+
+                // This lets us use just a width or height if desired
+                if("height" in requestData){
+                    requestData.height = parseInt(requestData.height)
+                }
+
+                if("width" in requestData){
+                    requestData.width = parseInt(requestData.width)
+                }
+
+                if(typeof(requestData.height) !== "undefined" && typeof(requestData.width) === "undefined" ){
```

Comment:
Hi,

It's not pedantry, it's good form. I actually have a few changes meant to
address the test errors; I just got distracted by work and haven't pushed
them.

Sorry about that!

-Robert

On Wed, Mar 16, 2016 at 3:13 AM, Mark Elphinstone-Hoadley <
notifications@github.com> wrote:

> In index.js
> <https://github.com/marksyzm/sickle/pull/2#discussion_r56308992>:
>
> > @@ -108,7 +108,30 @@ function resizeImage (filePath, requestData, data, options, cb) {
> >                  var width, height;
> >                  var size = metadata.size;
> >
> > -                if (options.scaleUp || (!options.scaleUp && ( size.width > requestData.width || size.height > requestData.height))) {
> > +
> > +                // This lets us use just a width or height if desired
> > +                if("height" in requestData){
> > +                    requestData.height = parseInt(requestData.height)
> > +                }
> > +
> > +                if("width" in requestData){
> > +                    requestData.width = parseInt(requestData.width)
> > +                }
> > +
> > +                if(typeof(requestData.height) !== "undefined" && typeof(requestData.width) === "undefined" ){
>
> Pedantry, I know, but would you mind putting typeof(requestData.height)
> into the preferred format of typeof requestData.height as it's an
> operator and not a function? Also, you probably should check that they
> contain numbers instead and reverse the operands as that is the only
> accepted type for height/width.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/marksyzm/sickle/pull/2/files/d1a169b9049a7c5c69fc1870f246aaf47b0685e1#r56308992>
>